src/commands/dev/socket.rs

```diff
@@ -42,26 +52,30 @@ pub async fn listen(session_id: &str) -> Result<(), failure::Error> {
             }
         })
     };
-    pin_mut!(keep_alive_to_ws, print_ws_messages);
-    future::select(keep_alive_to_ws, print_ws_messages).await;
-    Ok(())
+
+    // run the heartbeat and message printer in parallel
+    tokio::select! {
+        _ = keep_alive_to_ws => { Ok(()) }
+        _ = print_ws_messages => { Ok(()) }
+    }
 }
 
-async fn keep_alive(tx: futures::channel::mpsc::UnboundedSender<Message>) -> ! {
+async fn keep_alive(tx: mpsc::UnboundedSender<Message>) -> ! {
     let duration = Duration::from_millis(1000 * KEEP_ALIVE_INTERVAL);
-    let mut interval = time::interval(duration);
+    let mut delay = delay_for(duration);
 
     // this is set to 2 because we have already sent an id of 1 to enable the runtime
     // eventually this logic should be moved to the chrome-devtools-rs library
     let mut id = 2;
 
     loop {
-        interval.tick().await;
+        delay.await;
         let keep_alive_message = protocol::runtime::SendMethod::GetIsolateId(id.into());
         let keep_alive_message = serde_json::to_string(&keep_alive_message)
             .expect("Could not convert keep alive message to JSON");
         let keep_alive_message = Message::Text(keep_alive_message);
-        tx.unbounded_send(keep_alive_message).unwrap();
+        tx.send(keep_alive_message).unwrap();
         id += 1;
+        delay = delay_for(duration);
```

Comment:
this also mirrors how @ashleymichal implemented heartbeats in `wrangler tail` - functionally equivalent but moved for consistency.